patches/server/1039-Fix-NPE-in-SculkBloomEvent-world-access.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 20 Oct 2023 19:50:22 +0200
+Subject: [PATCH] Fix NPE in SculkBloomEvent world access
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/SculkSpreader.java b/src/main/java/net/minecraft/world/level/block/SculkSpreader.java
+index de90a216321f7d82310a0d1c915fefe64360534c..6010c6cc727cee90f61e15848bd111385c6d52fb 100644
+--- a/src/main/java/net/minecraft/world/level/block/SculkSpreader.java
++++ b/src/main/java/net/minecraft/world/level/block/SculkSpreader.java
+@@ -62,7 +62,7 @@ public class SculkSpreader {
+     private final int additionalDecayRate;
+     private List<SculkSpreader.ChargeCursor> cursors = new ArrayList();
+     private static final Logger LOGGER = LogUtils.getLogger();
+-    Level level; // CraftBukkit
++    public Level level; // CraftBukkit // Paper - package-private -> public
+ 
+     public SculkSpreader(boolean worldGen, TagKey<Block> replaceableTag, int extraBlockChance, int maxDistance, int spreadChance, int decayChance) {
+         this.isWorldGeneration = worldGen;
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
+index 6c378e3485fef4e8b8906c33a45a23e6f77e9cdd..35a052da5c0ba1f88d6788e3637502c427dd0b65 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
+@@ -35,6 +35,14 @@ public class SculkCatalystBlockEntity extends BlockEntity implements GameEventLi
+         catalystListener.level = level; // CraftBukkit
+     }
+ 
++    // Paper start
++    @Override
++    public void setLevel(Level level) {
++        super.setLevel(level);
++        this.catalystListener.getSculkSpreader().level = level;
```

Comment:
I would set the level directly via the `sculkSpreader` field instead of the getter. Just seems more safe that way in case the getter ever has more logic down the line or something.